sympy/physics/quantum/operator.py

```diff
@@ -186,6 +186,30 @@ def __mul__(self, other):
 
         return Mul(self, other)
 
+    # Issue #24153: Change the result Operator**0 to IdentityOperator() instead of 1
+    # Remarks:
+    # 1. Operators who know how to compute self**exp should override ._eval_power(exp),
+    #    since ._eval_power(exp) will be called from both Pow(op, exp) and op**exp.
+    #    In case self**exp == IdentityOperator(), ._eval_power(exp) should return
+    #    IdentityOperator() instead of 1.
+    # 2. Hooking into ._pow() instead would not give same effect as Pow.__new__()
+    #    doesn't call ._pow().
+    # 2. Limitation: Pow filters on base,exp pairs that it will compute by itself, and
+    #    will for almost all type of base return 1 if exp == 0 instead of returning
+    #    the neutral element of the multiplicative half group base belongs to.
+    # 3. This behavior of Pow can only be fixed by a modification of Pow itself.
+    # 4. But it can be modified for ** and pow(), by either overriding .__pow__() (which
+    #    however is considered bad practice) or better ._pow() for the Operator class,
+    #    handling exponent == 0 and passing on all other to the original super()._pow()
+    #    unmodified.
+
+    def _pow(self, exponent):
+        from sympy.core.singleton import S
+        if exponent == 0 or exponent == S.Zero:
```

Comment:
One of these should be enough as `S.Zero == 0`. (An option is to sympify `exponent`, unless it is known to be sympified already, which it may be as it is a "private" method, and check `exponent is S.Zero`.)

Reply:
I've learned to expect the unexpected, but thought that this code is cheaper than `sympify(exponent) == S.Zero`. May I rely on that exponent is sympified already, and why?

BTW: Is there a document that states design principles like these in sympy: Which functions may a user call - either interactive or in their code? Because that defines the 'line of defence' where non-sympified expressions may arrive, and sympification of arguments is required.  

 Or another design feature: how the function cache `cachit` works and should be used?